libraries/exoplayer_hls/src/main/java/androidx/media3/exoplayer/hls/playlist/HlsPlaylistParser.java

```diff
@@ -374,7 +411,39 @@ private static HlsMultivariantPlaylist parseMultivariantPlaylist(
         int roleFlags = isIFrameOnlyVariant ? C.ROLE_FLAG_TRICK_PLAY : 0;
         int peakBitrate = parseIntAttr(line, REGEX_BANDWIDTH);
         int averageBitrate = parseOptionalIntAttr(line, REGEX_AVERAGE_BANDWIDTH, -1);
+        String videoRange = parseOptionalStringAttr(line, REGEX_VIDEO_RANGE, variableDefinitions);
         String codecs = parseOptionalStringAttr(line, REGEX_CODECS, variableDefinitions);
+        String supplementalCodecsStrings =
+            parseOptionalStringAttr(line, REGEX_SUPPLEMENTAL_CODECS, variableDefinitions);
+        String supplementalCodecs = null;
+        String supplementalProfiles = null;  // i.e. Compatibility brand
+        if (supplementalCodecsStrings != null) {
+          String[] supplementalCodecsString = Util.split(supplementalCodecsStrings, ",");
+          if (!supplementalCodecsString[0].isEmpty()) {
+            // TODO: Support more than one element
+            String[] codecsAndProfiles = Util.split(supplementalCodecsString[0], "/");
+            supplementalCodecs = codecsAndProfiles[0];
+            if (codecsAndProfiles.length > 1) {
+              supplementalProfiles = codecsAndProfiles[1];
+            }
+          }
+        }
+        String videoCodecs = Util.getCodecsOfType(codecs, C.TRACK_TYPE_VIDEO);
+        if (isDolbyVisionFormat(videoRange, videoCodecs, supplementalCodecs, supplementalProfiles)) {
+          videoCodecs = supplementalCodecs != null ? supplementalCodecs : videoCodecs;
```

Comment:
(1) Just as my previous comment, it is used for profile 5.
(2) Yes, the codecs string here includes audio codec info too. For example, one of my test, the purpose of this code is to change
codecs = hvc1.2.4.L60.b0,ec-3
to
codecs = dvh1.08.01,ec-3

Reply:
(1) - Thanks, I missed the early return with `true`.

(2) Yes, that makes sense. The logic needs some minor adjustments though for the (unlikely) case of having multiple video codecs specified. I think it would currently add the same `videoCodecs` String multiple times.  What do you think about just building the codec string without any video codecs in the loop, and then appending them together at the end: `builder.length() > 0 ? videoCodecs + "," + builder.toString() : videoCodecs`? Even if that case never practically happens, it makes the code easier to understand and avoids confusion.

Optional: You could also create `Util.getCodecsWithoutType` (next to `Util.getCodecsOfType`) to make it even more readable.

Reply:
(2) It's fine. I'll update the code.

Reply:
Thanks!